tt_metal/llrt/rtoptions.cpp

```diff
@@ -32,15 +33,31 @@ const char* RunTimeDebugFeatureNames[RunTimeDebugFeatureCount] = {
 
 const char* RunTimeDebugClassNames[RunTimeDebugClassCount] = {"N/A", "worker", "dispatch", "all"};
 
+// Environment variable name for TT-Metal root directory
 constexpr auto TT_METAL_RUNTIME_ROOT_ENV_VAR = "TT_METAL_RUNTIME_ROOT";
-constexpr auto TT_METAL_KERNEL_PATH_ENV_VAR = "TT_METAL_KERNEL_PATH";
-// Set this var to change the cache dir.
-constexpr auto TT_METAL_CACHE_ENV_VAR = "TT_METAL_CACHE";
-// Used for demonstration purposes and will be removed in the future.
-// Env variable to override the core grid configuration
-constexpr auto TT_METAL_CORE_GRID_OVERRIDE_TODEPRECATE_ENV_VAR = "TT_METAL_CORE_GRID_OVERRIDE_TODEPRECATE";
-
-RunTimeOptions::RunTimeOptions() {
+
+namespace {
+// Helper function to normalize directory paths using std::filesystem
+std::string normalize_path(const char* path, const std::string& subdir = "") {
+    std::filesystem::path p(path);
+    if (!subdir.empty()) {
+        p /= subdir;
+    }
+    return p.lexically_normal().string();
+}
+
+// Helper function to check if environment variable value is "1" (enabled)
+inline bool is_env_enabled(const char* value) { return value && value[0] == '1'; }
```

Comment:
I don't think `inline` is needed. Compiler will do what it wants.

```suggestion
bool is_env_enabled(const char* value) { return value && value[0] == '1'; }
```